python/python/bystro/ancestry/train.py

```diff
@@ -376,39 +501,125 @@ def sample(cls) -> "RFCParamChoices":
         return cls(**kwargs)
 
 
-def _make_rfc(
-    train_Xpc: np.ndarray,
-    test_Xpc: np.ndarray,
-    train_y: pd.Series,
-    test_y: pd.Series,
+@dataclasses.dataclass(frozen=True)
+class AccuracyReport:
+    """Represent model accuracy scores at population and superpopulation levels."""
+
+    train_pop_accuracy: float
+    test_pop_accuracy: float
+    train_superpop_accuracy: float
+    test_superpop_accuracy: float
+
+
+def _compute_accuracy_report(
+    clf: sklearn.base.ClassifierMixin,
+    train_Xpc: pd.DataFrame,
+    test_Xpc: pd.DataFrame,
+    train_y: pd.DataFrame,
+    test_y: pd.DataFrame,
+) -> AccuracyReport:
+    train_yhat_pop_probs = clf.predict_proba(train_Xpc)
+    test_yhat_pop_probs = clf.predict_proba(test_Xpc)
+
+    train_yhat_pops = POPS[np.argmax(train_yhat_pop_probs, axis=1)]
+    test_yhat_pops = POPS[np.argmax(test_yhat_pop_probs, axis=1)]
+    train_yhat_superpops = superpop_predictions_from_pop_probs(train_yhat_pop_probs)
+    test_yhat_superpops = superpop_predictions_from_pop_probs(test_yhat_pop_probs)
+
+    return AccuracyReport(
+        accuracy_score(train_y.population, train_yhat_pops),
+        accuracy_score(test_y.population, test_yhat_pops),
+        accuracy_score(train_y.superpop, train_yhat_superpops),
+        accuracy_score(test_y.superpop, test_yhat_superpops),
+    )
+
+
+def make_rfc(
```

Comment:
Will the final version of train include multiple models (the RFC and the latest model)?

Reply:
We're going to aim for one model for now and see if a use case for a second emerges.  for now we're just publicizing the method so we can import it into a separate chip model training script later.